core/src/main/java/ch/unibas/dmi/dbis/polyphenydb/adapter/enumerable/EnumerableInterpretable.java

```diff
@@ -107,32 +108,35 @@ public EnumerableInterpretable copy( RelTraitSet traitSet, List<RelNode> inputs
 
     @Override
     public Node implement( final InterpreterImplementor implementor ) {
-        final Bindable bindable = toBindable( implementor.internalParameters, implementor.spark, (EnumerableRel) getInput(), EnumerableRel.Prefer.ARRAY, implementor.dataContext );
+        final Bindable bindable = toBindable(
+                implementor.internalParameters,
+                implementor.spark,
+                (EnumerableRel) getInput(),
+                EnumerableRel.Prefer.ARRAY,
+                implementor.dataContext.getTransaction() );
         final ArrayBindable arrayBindable = box( bindable );
         final Enumerable<Object[]> enumerable = arrayBindable.bind( implementor.dataContext );
         return new EnumerableNode( enumerable, implementor.compiler, this );
     }
 
 
-    public static Bindable toBindable( Map<String, Object> parameters, SparkHandler spark, EnumerableRel rel, EnumerableRel.Prefer prefer, DataContext dataContext ) {
+    public static Bindable toBindable( Map<String, Object> parameters, SparkHandler spark, EnumerableRel rel, EnumerableRel.Prefer prefer, Transaction transaction ) {
         EnumerableRelImplementor relImplementor = new EnumerableRelImplementor( rel.getCluster().getRexBuilder(), parameters );
 
         final ClassDeclaration expr = relImplementor.implementRoot( rel, prefer );
         String s = Expressions.toString( expr.memberDeclarations, "\n", false );
 
-        if ( PolyphenyDbPrepareImpl.DEBUG ) {
+        if ( RuntimeConfig.DEBUG.getBoolean() ) {
             Util.debugCode( System.out, s );
         }
 
-        if ( dataContext != null && dataContext.getTransaction().isAnalyze() ) {
-            InformationManager queryAnalyzer = dataContext.getTransaction().getQueryAnalyzer();
+        if ( transaction.isAnalyze() ) {
```

Comment:
Do we know for sure that `transaction` is never `null`? If not, should we keep the null check?

Reply:
There was no null check for transaction, only for dataContext. But you are right: a null check won't hurt. I will add one.

Reply:
There was no null check for transaction, only for dataContext. But you are right: a null check won't hurt. I will add one.